src/init.cpp

```diff
@@ -1024,6 +1024,22 @@ bool AppInitParameterInteraction(const ArgsManager& args)
     if (args.GetBoolArg("-peerbloomfilters", DEFAULT_PEERBLOOMFILTERS))
         nLocalServices = ServiceFlags(nLocalServices | NODE_BLOOM);
 
+    if (args.IsArgSet("-test")) {
+        if (chainparams.GetChainType() != ChainType::REGTEST) {
+            return InitError(Untranslated("-test=<option> should only be used in functional tests"));
```

Comment:
nit: I think this makes the error slightly clearer
```suggestion
            return InitError(Untranslated("-test=<option> can only be used when on regtest"));
```

Reply:
included in https://github.com/bitcoin/bitcoin/pull/29636